fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/PreparableModelLoadingPlugin.java

```diff
@@ -63,4 +74,11 @@ interface DataLoader<T> {
 		 */
 		CompletableFuture<T> load(ResourceManager resourceManager, Executor executor);
 	}
+
+	/**
+	 * Bundles a {@link PreparableModelLoadingPlugin} with its corresponding {@link DataLoader}
+	 * for retrieval through {@link #getAll()}.
+	 */
+	record Holder<T>(DataLoader<T> loader, PreparableModelLoadingPlugin<T> plugin) {
```

Comment:
Generally we dont expose a record as API, I would suggest using an interface that has a `DataLoader<T> loader()` and `PreparableModelLoadingPlugin<T> plugin()`. You can then impliment that interface with a record like this one but in impl.